tutorials/Land/Soil/Coupled/equilibrium_test.jl

```diff
@@ -424,101 +391,74 @@ boundaries = [
-    T = iaux[:, T_ind, :][:]
-    K∇h_vert = igrads[:, K∇h_vert_ind, :][:]
-    κ∇T_vert = igrads[:, κ∇T_vert_ind, :][:]
-    all_vars = Dict{String, Array}(
-        "t" => [t],
-        "ϑ_l" => ϑ_l,
-        "T" => T,
-        "K∇h_vert" => K∇h_vert,
-        "κ∇T_vert" => κ∇T_vert,
-    )
-    all_data[step[1]] = all_vars
 
-    step[1] += 1
+callback = GenericCallbacks.EveryXSimulationTime(every_x_simulation_time) do
+    dons = dict_of_nodal_states(solver_config, state_types; interp = true)
+    push!(all_data, dons)
+    push!(time_data, gettime(solver_config.solver))
     nothing
 end;
 
-
 # # Run the integration
 ClimateMachine.invoke!(solver_config; user_callbacks = (callback,));
 
 
 # Get the final state and create plots:
-t = ODESolvers.gettime(solver_config.solver)
-iQ, iaux, igrads = interpolate_variables((Q, aux, grads), intrp_brck)
-
-ϑ_l = iQ[:, ϑ_l_ind, :][:]
-T = iaux[:, T_ind, :][:]
-K∇h_vert = igrads[:, K∇h_vert_ind, :][:]
-κ∇T_vert = igrads[:, κ∇T_vert_ind, :][:]
-all_vars = Dict{String, Array}(
-    "t" => [t],
-    "ϑ_l" => ϑ_l,
-    "T" => T,
-    "K∇h_vert" => K∇h_vert,
-    "κ∇T_vert" => κ∇T_vert,
+dons = dict_of_nodal_states(solver_config, state_types; interp = true)
+push!(all_data, dons)
+push!(time_data, gettime(solver_config.solver));
+
+# Get z-coordinate
+z = get_z(solver_config.dg.grid; rm_dupes = true);
+
+# Let's export a plot of the initial state
+output_dir = @__DIR__;
+
+mkpath(output_dir);
+
+export_plot(
+    z,
+    all_data,
+    ("soil.water.ϑ_l",),
+    joinpath(output_dir, "eq_moisture_plot.png");
+    xlabel = "ϑ_l",
+    ylabel = "z (cm)",
+    time_data = time_data,
 )
-all_data[n_outputs] = all_vars
-iz = iaux[:, z_ind, :][:]
-
-t = [all_data[k]["t"][1] for k in 1:n_outputs]
-t = ceil.(Int64, t ./ 60)
-
-ϑ_plot =
-    plot(initial_state["ϑ_l"], z, label = "t = 0", ylabel = "z", xlabel = "ϑ_l")
-plot!(all_data[1]["ϑ_l"], iz, label = "t = 0.75 days")
-plot!(all_data[2]["ϑ_l"], iz, label = "t = 1.5 days")
-plot!(all_data[3]["ϑ_l"], iz, label = "t = 2.25 days")
-plot!(all_data[4]["ϑ_l"], iz, label = "t = 3 days")
-
-
-K∇h_z_plot = plot(
-    all_data[1]["K∇h_vert"],
-    iz,
-    label = "0.75 days",
-    xlabel = "K∇h_z (m/s)",
-)
-plot!(all_data[2]["K∇h_vert"], iz, label = "1.5 days")
-plot!(all_data[3]["K∇h_vert"], iz, label = "2.25 days")
-plot!(all_data[4]["K∇h_vert"], iz, label = "3 days")
-plot!(legend = :bottomleft)
-plot(ϑ_plot, K∇h_z_plot)
-savefig("eq_moisture_plot.png")
 # ![](eq_moisture_plot.png)
 
-T_plot =
-    plot(initial_state["T"], z, label = "t = 0", ylabel = "z", xlabel = "T (K)")
-plot!(all_data[1]["T"], iz, label = "t = 0.75 days")
-plot!(all_data[2]["T"], iz, label = "t = 1.5 days")
-plot!(all_data[3]["T"], iz, label = "t = 2.25 days")
-plot!(all_data[4]["T"], iz, label = "t = 3 days")
-plot!(legend = :bottomright)
-
-κ∇T_z_plot = plot(
-    all_data[1]["κ∇T_vert"],
-    iz,
-    label = "0.75 days",
-    xlabel = "κ∇T_z (W/m^2)",
+export_plot(
```

Comment:
This plot looks incorrect, not sure why?

Reply:
oops, maybe comment on the wrong line - the gradient of the head one looks funny

Reply:
I think I need to rebase / retrigger the build to get some recent fixes.

Reply:
@kmdeck I think the plots are looking as expected, now.